src/main/java/org/spongepowered/api/resourcepack/ResourcePacks.java

```diff
@@ -0,0 +1,46 @@
+package org.spongepowered.api.resourcepack;
+
+import com.google.common.base.Optional;
+
+import java.io.FileNotFoundException;
+import java.net.URL;
+
+public final class ResourcePacks {
+
+    public static final ResourcePackFactory factory = null;
+
+    /**
+     * Creates a {@link ResourcePack} from a URL and tries to download and hash
+     * it.
+     * 
+     * @param url The URL to look in.
+     * @return A ResourcePack with the specified URL.
+     * @throws FileNotFoundException If a valid resourcepack could not be
+     *             downloaded from the URL.
+     */
+    public static ResourcePack fromURL(URL url) throws FileNotFoundException {
```

Comment:
@Zidane I was told to make this class to be analogous to Messages and MessageFactory.